File: MemoryCheck/app/src/main/java/com/example/velm/memorycheck/MainActivity.java
package com.example.velm.memorycheck;

import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    /*@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView tv = new TextView(this);
        tv.setText("Hello");
        setContentView(tv);

        Context context = this;

        if(context != null){

            Log.d("CON",context.getPackageName().toString());
        }

        Context context2 = getApplicationContext();

        if(context2 != null){
            Log.d("context2",context2.getPackageName().toString());
        }
    }*/

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("OnResume","OnResume");
    }
}

File: CoroutinesExample/app/src/main/java/com/example/coroutinesexample/MainActivity.kt
package com.example.coroutinesexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {

    lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        setupViewModel()


    }

    private fun setupViewModel() {
        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        viewModel.users.observe(this, Observer {
            it?.let {

                text.text = it.toString()
                Log.d("TAG", "setupViewModel: ${it}")
            }
        })

        viewModel.fetchUsers()

    }
}